src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -350,16 +354,47 @@ private void initializeFromClientConfig(JedisClientConfig config) {
         auth(credentialsProvider);
       }
 
+      List<CommandArguments> fireAndForgetMsg = new ArrayList<>();
+
       int dbIndex = config.getDatabase();
       if (dbIndex > 0) {
-        select(dbIndex);
+        fireAndForgetMsg.add(new CommandArguments(Command.SELECT).add(Protocol.toByteArray(dbIndex)));
       }
+
       String clientName = config.getClientName();
       if (clientName != null) {
-        // TODO: need to figure out something without encoding
-        clientSetname(clientName);
+        fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETNAME).add(clientName));
+      }
+
+      String libName = JedisMetaInfo.getArtifactId();
+      if (libName != null) {
+        fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETINFO)
+            .add(ClientAttrOption.LIB_NAME.getRaw()).add(libName));
+      }
+
+      String libVersion = JedisMetaInfo.getVersion();
+      if (libVersion != null) {
+        fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETINFO)
+            .add(ClientAttrOption.LIB_VER.getRaw()).add(libVersion));
+      }
+
+      for (CommandArguments arg : fireAndForgetMsg) {
+        sendCommand(arg);
       }
 
+      List<Object> objects = getMany(fireAndForgetMsg.size());
+      for (Object obj : objects) {
+        if (obj instanceof JedisDataException) {
+          JedisDataException e = (JedisDataException)obj;
+          String errorMsg = e.getMessage().toUpperCase();
+          if (errorMsg.contains("UNKNOWN") ||
+              errorMsg.contains("NOAUTH")) {
```

Comment:
@sazzad16 Here, we need to discuss: Jedis allows the following usage

```
Jedis jedis = new Jedis(host, port);
jedis.auth("xxx");
```

Since `auth` is run in the second step, the `client setinfo` command here will necessarily return a `NOAUTH` error, so I ignored `NOAUTH` here, and I modified the `testCredentialsProvider` test.

Reply:
@yangbodong22011 Agree. One idea is to enforce the constructor(s) to be sufficient for a connection. So NOAUTH won't be tolerated here. We can target this change for Jedis 5.

Reply:
OK，so here we keep filtering NOAUTH.

Reply:
Yes, but add a comment maybe?
```suggestion
              errorMsg.contains("NOAUTH")) { // TODO: not filter out NOAUTH
```